detectron2/modeling/roi_heads/fast_rcnn.py

```diff
@@ -547,7 +547,7 @@ def predict_boxes(self, predictions, proposals):
         if not len(proposals):
             return []
         _, proposal_deltas = predictions
-        num_prop_per_image = [len(p) for p in proposals]
+        num_prop_per_image = [p.proposal_boxes.tensor.size(0) for p in proposals]
```

Comment:
Changes in this file are not tested. Are they still needed?

Reply:
I've now added tests for them.